source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -50,11 +53,112 @@ void Filter::onRead() {
 
 void Filter::onReadWorker() {
   Network::ConnectionSocket& socket = cb_->socket();
-  std::string proxy_line;
-  if (!readLine(socket.fd(), proxy_line)) {
+  PpHeader hdr;
+  hdr.valid = false;
```

Comment:
Instead of using a valid boolean here, how about passing an absl::optional that can get filled and then check if it's filled?